backend/src/main/java/com/bakdata/conquery/models/events/CBlock.java

```diff
@@ -139,30 +139,31 @@ else if (treeConcept.countElements() == 1) {
 
 
 			try {
-				// Events without values are assigned to the root
-				// Events can also be filtered, allowing a single table to be used by multiple connectors.
-				if (column != null && !bucket.has(event, column)) {
-					mostSpecificChildren[event] = treeConcept.getPrefix();
-					continue;
-				}
-
 				String stringValue = "";
 
-				if (column != null) {
+				final boolean has = column != null && bucket.has(event, column);
+
+				if (column != null && has) {
 					stringValue = bucket.getString(event, column);
 				}
 
+				// Events can also be filtered, allowing a single table to be used by multiple connectors.
 				// Lazy evaluation of map to avoid allocations if possible.
 				// Copy event for closure.
 				final int _event = event;
 				final CalculatedValue<Map<String, Object>> rowMap = new CalculatedValue<>(() -> bucket.calculateMap(_event));
 
-
 				if (connectorCondition != null && !connectorCondition.matches(stringValue, rowMap)) {
 					mostSpecificChildren[event] = Connector.NOT_CONTAINED;
 					continue;
 				}
 
+				// Events without values are assigned to the root
+				if (column != null && !has) {
+					mostSpecificChildren[event] = treeConcept.getPrefix();
+					continue;
+				}
```

Comment:
Heißt das dann nicht auch wenn eine entitiy nie ein event für ein concept hatte, dass wenn man den wurzelknoten reinzieht, die entitiy dann auftaucht?

Reply:
sie muss werte in der Tabelle haben, das ist auch das Ziel des PRs, dass auf der Wurzel alle einträge rauskommen, die in der Tabelle einen Eintrag haben; aber nicht umbedingt einen Wert in column haben.

Reply:
bzw, des vorgegangenen PRs, hier war noch die Reihenfolge von Filter und root assignment falsch.